sdk/avs/Azure.ResourceManager.Avs/src/Customized/ArmAvsModelFactory.cs

```diff
@@ -45,13 +46,198 @@ public static partial class ArmAvsModelFactory
+        /// <param name="secondaryCircuit">
+        /// A secondary expressRoute circuit from a separate AZ. Only present in a
+        /// stretched private cloud
+        /// </param>
+        /// <param name="nsxPublicIPQuotaRaised">
+        /// Flag to indicate whether the private cloud has the quota for provisioned NSX
+        /// Public IP count raised from 64 to 1024
+        /// </param>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public static AvsPrivateCloudData AvsPrivateCloudData(
+            ResourceIdentifier id,
+            string name,
+            ResourceType resourceType,
+            ResourceManager.Models.SystemData systemData,
+            IDictionary<string, string> tags,
+            AzureLocation location,
+            string sku,
+            ManagedServiceIdentity identity,
+            AvsManagementCluster managementCluster,
+            InternetConnectivityState? internet,
+            IEnumerable<SingleSignOnIdentitySource> identitySources,
+            PrivateCloudAvailabilityProperties availability,
+            CustomerManagedEncryption encryption,
+            IEnumerable<string> extendedNetworkBlocks,
+            AvsPrivateCloudProvisioningState? provisioningState,
+            ExpressRouteCircuit circuit,
+            AvsPrivateCloudEndpoints endpoints,
+            string networkBlock,
+            string managementNetwork,
+            string provisioningNetwork,
+            string vMotionNetwork,
+            string vCenterPassword,
+            string nsxtPassword,
+            string vCenterCertificateThumbprint,
+            string nsxtCertificateThumbprint,
+            IEnumerable<ResourceIdentifier> externalCloudLinks,
+            ExpressRouteCircuit secondaryCircuit,
+            NsxPublicIPQuotaRaisedEnum? nsxPublicIPQuotaRaised)
+        {
+            return new AvsPrivateCloudData(
+                id,
+                name,
+                resourceType,
+                systemData,
+                tags,
+                location,
+                sku != null ? new AvsSku(sku) : null,
+                identity,
+                managementCluster,
+                internet,
+                identitySources?.ToList(),
+                availability,
+                encryption,
+                extendedNetworkBlocks?.ToList(),
+                provisioningState,
+                circuit,
+                endpoints,
+                networkBlock,
+                managementNetwork,
+                provisioningNetwork,
+                vMotionNetwork,
+                vCenterPassword,
+                nsxtPassword,
+                vCenterCertificateThumbprint,
+                nsxtCertificateThumbprint,
+                externalCloudLinks?.ToList(),
+                secondaryCircuit,
+                nsxPublicIPQuotaRaised,
+                null,
+                null,
+                null
+            );
+        }
```

Comment:
the methods here should return the generated version of this method, passing parameters by names, such as
```suggestion
        [EditorBrowsable(EditorBrowsableState.Never)]
        public static AvsPrivateCloudData AvsPrivateCloudData(
            ResourceIdentifier id,
            string name,
            ResourceType resourceType,
            ResourceManager.Models.SystemData systemData,
            IDictionary<string, string> tags,
            AzureLocation location,
            string sku,
            ManagedServiceIdentity identity,
            AvsManagementCluster managementCluster,
            InternetConnectivityState? internet,
            IEnumerable<SingleSignOnIdentitySource> identitySources,
            PrivateCloudAvailabilityProperties availability,
            CustomerManagedEncryption encryption,
            IEnumerable<string> extendedNetworkBlocks,
            AvsPrivateCloudProvisioningState? provisioningState,
            ExpressRouteCircuit circuit,
            AvsPrivateCloudEndpoints endpoints,
            string networkBlock,
            string managementNetwork,
            string provisioningNetwork,
            string vMotionNetwork,
            string vCenterPassword,
            string nsxtPassword,
            string vCenterCertificateThumbprint,
            string nsxtCertificateThumbprint,
            IEnumerable<ResourceIdentifier> externalCloudLinks,
            ExpressRouteCircuit secondaryCircuit,
            NsxPublicIPQuotaRaisedEnum? nsxPublicIPQuotaRaised)
        {
            return AvsPrivateCloudData(
                id: id,
                name: name,
                resourceType: resourceType,
                systemData: systemData,
                // others should follow the same pattern
                tags,
                location,
                sku != null ? new AvsSku(sku) : null,
                identity,
                managementCluster,
                internet,
                identitySources?.ToList(),
                availability,
                encryption,
                extendedNetworkBlocks?.ToList(),
                provisioningState,
                circuit,
                endpoints,
                networkBlock,
                managementNetwork,
                provisioningNetwork,
                vMotionNetwork,
                vCenterPassword,
                nsxtPassword,
                vCenterCertificateThumbprint,
                nsxtCertificateThumbprint,
                externalCloudLinks?.ToList(),
                secondaryCircuit,
                nsxPublicIPQuotaRaised,
                null,
                null,
                null
            );
        }
```

Reply:
This change is based on the fact that we will always ensure the methods in model factory will not break because they are public.
Therefore if we call the public method here, no matter how many changes it would receive in the future, this invocation will always work.
In this way, we do not have to add the internal ctor to the customized file every time it changes. (see other comments that I left to remove the internal ctors in customizations)